src/execution_plan/execution_plan.c

```diff
@@ -709,8 +715,8 @@ static void _ExecutionPlanSegment_ConvertClause(GraphContext *gc, AST *ast,
 	// Because 't' is set using the offsetof() call, it cannot be used in switch statements.
 	if(t == CYPHER_AST_MATCH) {
 		// Only add at most one set of traversals per plan. TODO Revisit and improve this logic.
-		if(ExecutionPlan_LocateFirstOp(plan->root, OPType_NODE_BY_LABEL_SCAN) ||
-		   ExecutionPlan_LocateFirstOp(plan->root, OPType_ALL_NODE_SCAN)) {
+		if(plan->root && (ExecutionPlan_LocateFirstOp(plan->root, OPType_NODE_BY_LABEL_SCAN) ||
```

Comment:
Is it possible for `plan->root` to be `NULL` at this point?

Reply:
Absolutely - any query that begins with a `MATCH` clause would have a NULL root right now. The condition change here is to avoid excessive calls to `ExecutionPlan_LocateFirstOp` - an optimization, not a fix.